src/Simulation.jl

```diff
@@ -47,7 +47,7 @@ struct SystemSimulation{Q,B<:AbstractODESolution,X,Y,Z}
 end
 
 function SystemSimulation(sol,domains,interfaces,p)
-    sims = Tuple([Simulation(sol,domain) for domain in domains])
+    sims = Tuple([Simulation(sol,domain,interfaces,p) for domain in domains])
```

Comment:
Do we want each Simulation object to have all of the interface objects in the system?

Reply:
You're right, they probably only need to carry interfaces that are connected to them